### PR TITLE
[react, react-native] Fix double connection issue, add ConnectEmbed component

### PR DESCRIPTION
**Pull request by MananTank**

### CODE REVIEW

File: packages/react-core/src/core/types/wallet.ts [hunk 1]
```diff
@@ -156,9 +173,75 @@ export type ConnectUIProps<I extends WalletInstance = WalletInstance> = {
   /**
    * Called when the wallet is connected but it's
    * part of another wallet's connection flow.
-   * @param walleInstance - the instance of the connected wallet
+   * @param walletInstance - the instance of the connected wallet
+   */
+  onLocallyConnected?: (walletInstance: WalletInstance) => void;
```
Comment (MananTank) [inline packages/react-core/src/core/types/wallet.ts:178]: This was added for react-native, with the new API - we don't need it anymore

**Comment by iketw** [comment]
@MananTank could you expand on what's the main change in this PR? Maybe a doc will help, this one is a big PR.

File: packages/react-core/src/core/providers/thirdweb-wallet-provider.tsx [hunk 2]
```diff
@@ -285,20 +285,220 @@ export function ThirdwebWalletProvider(
...
+    setConnectedWallet,
+    switchChain,
+    connectWallet,
+    disconnectWallet,
+  } = useWalletConnectionSetup(walletSetupData, {
+    connectionStatus: "unknown",
+  });
+
+  const personalWalletConnection = useWalletConnectionSetup(walletSetupData, {
+    connectionStatus: "disconnected",
```
Comment (MananTank) [inline packages/react-core/src/core/providers/thirdweb-wallet-provider.tsx:458]: TODO: remove this arg - always start with connectionStatus "unknown"

**Pull request merged**